packages/now-build-utils/src/fs/run-user-scripts.ts

```diff
@@ -349,20 +353,50 @@ export async function runPipInstall(
   );
 }
 
+export interface GetScriptNameOptions {
+  allowPlatformName?: boolean;
+}
+
+export function getScriptName(
+  baseScriptName: string,
+  pkg?: Pick<PackageJson, 'scripts'>,
+  opts?: GetScriptNameOptions
+): string | null {
+  if (pkg && pkg.scripts) {
+    if (opts && opts.allowPlatformName) {
+      const vercelScriptName = `vercel-${baseScriptName}`;
+      if (vercelScriptName in pkg.scripts) {
+        return vercelScriptName;
+      }
+
+      const nowScriptName = `now-${baseScriptName}`;
+      if (nowScriptName in pkg.scripts) {
+        return nowScriptName;
+      }
+    }
+
+    if (baseScriptName in pkg.scripts) {
+      return baseScriptName;
+    }
+  }
+
+  return null;
+}
+
+export type RunPackageJsonScriptOptions = GetScriptNameOptions & SpawnOptions;
+
 export async function runPackageJsonScript(
   destPath: string,
-  scriptName: string,
-  spawnOpts?: SpawnOptions
+  baseScriptName: string,
```

Comment:
I think this is going to cause problems for Community Builders which might not want to try 3 different script names.

I think we should leave `runPackageJsonScript` unchanged and instead call `getScriptName()` from `@vercel/node` and `@vercel/next` to have more fine-grained control. For example `@vercel/node` shouldn't execute `build`.

Reply:
Ya I made it be opt-in now by setting `allowPlatformName: true` in the options object.

Reply:
Oh that is an interesting way to do it? Seems a little confusing as an API though. Its not clear what `allowBaseName` or `allowPlatformName` options do without reading the source code.

When we introduced Node version selection, we added a new function `getSpawnOptions()` that Community Runtimes had to explicitly call.

I think we should do the same thing here. Export a new function that an array of options and return the first one: 

```js
// "@vercel/next"
const script = getScriptName(pkgJson, ['vercel-build', 'now-build', 'build']);
if (script) {
  await runPackageJsonScript(dest, script, opts);
}
```


```js
// "@vercel/node"
const script = getScriptName(pkgJson, ['vercel-build', 'now-build']);
if (script) {
  await runPackageJsonScript(dest, script, opts);
}
```

I think this would reduce the changes to `runPackageJsonScript()` and also make `getScriptName()` very simple.